Working on updating Manual

Int-Circuit opened the pull request:
### Why
The QelectroTech manual hasn't been updated for V0.9 and lacks any developer info (build/test the app). This MR hopes to correct this.

### Summary
This MR has three axes:
### Simplify

- Switched from using bootstap4 theme to furo sphinx theme. (Reason is that the former isn't updated anymore. The latter is updated, is used for the python + pip doc and has built-in support for dark theme.)
- Use sphinx-design extension to simplify appearance of welcome page.
- Create a (proprer) directory structure to switch from a messy structure to an organized one (especially for images) 
- Ported old documentation contents to online doc (https://github.com/qelectrotech/qelectrotech-source-mirror/pull/412)

### Update

- Added Dev docs on how to build the app 
- Added GitHub CI to automaticaly build, publish and licence files on push/merge/ pull request to branch. Using [Sphinx-note's page action](https://github.com/sphinx-notes/pages) for page deployment and custom action for licensing file (see https://github.com/Int-Circuit/qelectrotech-doc/pull/2 for an example of it)
- Added app changelog in developper/app_changelog.rst + Documentation changelog
- Created doc hub on homepage instead of going directly in user doc
- Added [automatic Doxygen doc support](https://github.com/qelectrotech/qelectrotech-source-mirror/pull/389)

### Maintain

- Update User docs
- Update low-res icons/images
- Add spellchecking system on deploy 

### Bugfix
- Fix https://github.com/qelectrotech/qelectrotech-doc/issues/1
- Fix https://github.com/qelectrotech/qelectrotech-doc/issues/2

Int-Circuit committed:
first commit

Int-Circuit committed:
starting to fix links

Int-Circuit commented:
I sadly cannot (yet) replicate the presentation on the user par of the manual. Would a presentation change be a no-go?

![image](https://github.com/user-attachments/assets/3088f24e-1489-4187-9c92-22469b229694)

scorpio810 commented:
I'm wait to push your pr.
Thanks @Integrat-edCircuit ;-)

scorpio810 commented:
if documentation were easier to translate, it would be a great improvement

Int-Circuit commented:
> if documentation were easier to translate, it would be a great improvement

It requires sadly to have the doc being rewritten in the corresponding language. Not an easy process. If I have some time, I'm gonna sse if there is a possibility to do som CI magic to use a libretranslate instance to do the job

scorpio810 commented:
https://qelectrotech.org/forum/viewtopic.php?pid=15771#p15771

Int-Circuit commented:
Interesting, I will see if it's possible to interface both of the systems

Le 20 juillet 2024 17:03:16 GMT+02:00, Laurent Trinques ***@***.***> a écrit :
>https://qelectrotech.org/forum/viewtopic.php?pid=15771#p15771
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/qelectrotech/qelectrotech-doc/pull/3#issuecomment-2241176729
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

Int-Circuit commented:
> I sadly cannot (yet) replicate the presentation on the user par of the manual. Would a presentation change be a no-go?
> 
> ![image](https://private-user-images.githubusercontent.com/136177157/350302775-3088f24e-1489-4187-9c92-22469b229694.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE2Nzc1ODEsIm5iZiI6MTcyMTY3NzI4MSwicGF0aCI6Ii8xMzYxNzcxNTcvMzUwMzAyNzc1LTMwODhmMjRlLTE0ODktNDE4Ny05YzkyLTIyNDY5YjIyOTY5NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyMlQxOTQxMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YWIwY2IwYzhmMTZlYTI2OGJjMTVlNjlkYmY2ZjgzMTM1NWYxOGJiODgxYmIzMmM2M2NhYTliOTEwZWY3ZmU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.S9lUIC-Ub8-xqHd9pooWwCyNubHQOBBlV73z1Qu2d00)

I've got it working by using the [sphinx-design extension](https://sphinx-design.readthedocs.io/en/furo-theme/index.html) so no custom html that could break. Yay!

![image](https://github.com/user-attachments/assets/d46a9fb8-90e0-41b0-9cc5-6c9222fd3d60)

Int-Circuit committed:
cleaning directories + grid view welcome screen user doc

removed sphinx-autobuild as it wasn't usefull for our case

Int-Circuit committed:
fixed old links + rework of folio 256x256 icon

Int-Circuit committed:
added more icons + cleanup

added more icons

Int-Circuit commented:
@scorpio810 is there something that changed meaningfully between 0.8 (last up to date version of the doc) and 0.9 that I would need to add/edit?

Int-Circuit committed:
Update build.rst

scorpio810 commented:
> @scorpio810 is there something that changed meaningfully between 0.8 (last up to date version of the doc) and 0.9 that I would need to add/edit?

@Integrat-edCircuit :See
https://qelectrotech.org/forum/viewtopic.php?id=1953
https://qelectrotech.org/forum/viewtopic.php?id=2379

https://qelectrotech.org/forum/viewtopic.php?pid=17250#p17250

scorpio810 commented:
https://github.com/qelectrotech/qelectrotech-source-mirror/blob/master/ChangeLog

scorpio810 commented:
yes, I know it's a lot of work to edit, but you don't have to, otherwise thank you very much in advance.

Int-Circuit commented:
> yes, I know it's a lot of work to edit, but you don't have to, otherwise thank you very much in advance.

Thx for linking me to that,  regarding what I was talking onto our  forum thread, what do you think about

Int-Circuit committed:
created new "interface" icon + corrected page name from "Qelectrotech" to "QElectroTech" to respect syntax of old manual

Int-Circuit committed:
directory cleanup + interface.png icon change

directory cleanup

Int-Circuit committed:
starting to add connector chapter

fixed conductor chapter

Int-Circuit committed:
added report icon + interface chapter

Int-Circuit committed:
Fixing dead links + automating Licensing thanks to REUSE tool

Int-Circuit committed:
reorganising images paths

Int-Circuit committed:
Adding REUSE Github Action

fixed name to launch action

changed activation of reuse action on push and PR on main + rewriting readME

Removed cached LICENSE folder as REUSE tool automatically downloads it.

fixed path in action

Int-Circuit committed:
added flag to reuse so that unreconnized files are ignored

Files are now REUSE-compliant

Int-Circuit committed:
adding recursive flag

Files are now REUSE-compliant

changed dependency step in action

added exclusion for licences folder

fixed non inclusion of licence folder for REUSE github action

Files are now REUSE-compliant

removed old licence message

fixed LICENCES dir inclusion

Files are now REUSE-compliant

Int-Circuit committed:
starting to add grid to dev doc + added changelog page

fixed typo in action

fixed git add in action

fixed errors

added flag to command

Int-Circuit committed:
separeted CIs + using action to push to GH pages

fixed wrong path in page action

REUSE fix

removed typo

added changes handeling to CI

making reuse job fail not the all action

typo in action

Int-Circuit commented:
Automatic page building now works! as for the REUSE action if you see recent failed deployments, it is normal it fails if no file have been changed

Int-Circuit committed:
Added Forum + Home button to documentation

added QET homepage

error in QET homepage syntax

Int-Circuit committed:
added app changelog

Update gh-reuse.yml to correctly account if no changes

Update gh-reuse.yml to take note of continue-on-fail on previous step

fixed position of if

Update gh-reuse.yml

Changed from activating on push to activating on pull request + removed unnecessary if no change step as the continue-if-fail instruction allow to skip got step if source files haven't had a change

added po files to translate + towncrier config to auto generate changelog of doc

added PO files

Int-Circuit commented:
@scorpio810 what is the ACTUAL version number of the doc?

scorpio810 commented:
There was never a version number for the documentation, it was just based on version 0.7 of QET.

Int-Circuit commented:
OK. Im currently implementing an autochangelog system for the doc + source code. Does it makes sense to say that the doc was at version 0.1 for qet 0.7?

scorpio810 commented:
Yes it's make sense.

Int-Circuit committed:
Reworking Icon dir

removed autochangelog to future MR + rework image dir

fixed icon error in chapters

Int-Circuit committed:
Finished updating print chapter

Int-Circuit committed:
finished element collection

testing changes to fix CI

excluding conf.py file

fixed path of source files in codespell step

added fix for codespell

added Test of path for codespell

changes to codespell

fixing errors in CI

Int-Circuit committed:
added preference chapter + revert codespell for now as info is needed from devs regarding action

fixing typos and formatting

Int-Circuit committed:
finished report chapter

Int-Circuit committed:
added folio chapter

added checkout

reenabled typo checker

skip svg files

fixed path to check spelling

fixed typos in previous chapters

Int-Circuit committed:
- prevent page deployment to be stopped if codespell reports errors. Spell errors still shown but deploy anyway
- Continue to fix errors in docs

Switch to always in action

removed always

use correct continue on error parameter

fixing quotes in skipped files param

fix quotes

fixing globs

Int-Circuit committed:
finished folio chapter

Int-Circuit committed:
start implement schema + project

Int-Circuit committed:
finished refactoring of all chapters to new code/dir format + fixing cosmetics

removed old blockquote from doc + some spelling fixes.

remove broken translations

remove more translations

added doxygen job to page setup

added sanity check

commented prod command

added wait on page job

corrected path on artifact

added test doxygen

forgot to point to test branch

use git switch instead of checkout

switch correct directory

fix artifact directory

check xml dir

return to origin path

added absolute path

dir check

switch to prebuilt action

remove workspace path

correct doxyfile placement

fix file name

added file passing to jobs

Int-Circuit commented:
@scorpio810 Could you perhaps ask some contributors like @plc-user or others to check if the docs are missing some implemented features pls?

plc-user commented:
Hello @Int-Circuit!

Most of my commits work in the background not to be seen by users, but some are quite visible:
- implement settable point-size of grid (1 .. 5)
- element editor: add mirror (shortcut: &lt;M&gt;) and flip (shortcut: &lt;F&gt;) for all primitives
- element-editor: add rotation with smaller increments (5 degrees) for "line", "polygon", "text" and "dynamic_text" with shortcut &lt;Ctrl&gt;+&lt;Space&gt;
- element-editor: fix rotation by 90° of primitives (shortcut: &lt;Space&gt;)
- element-editor: copy-and-paste: set paste-position to meaningful values
- export: set maximum width / height according limitations in QPainter to 32767 for Pixel-Images and 100000 for svg/dxf (was set to 10000 before globally for all types of export)
- use QET_ElementScaler from Element-Editor - GUI
- add "Company-Collection" and "Company-Titleblocks" (can be used like customer-collection and customer-titleblocks)

All these changes are for the version 0.10-dev of QET!

scorpio810 commented:
Hello @Int-Circuit!
I've compiled the html pages of the manual and made them available here for QET's users to comment on.:
[Manual_new](https://download.qelectrotech.org/qet/manuals/html/index.html)

Int-Circuit commented:
Hello Laurent,

Thanks for that,

The doc isn't yet completelly ironed out but that's a start and this step will help with that.
When it's done I will give you help on how to setup the automatic deployment of the doc with github.

The only thing we would need would be a subdomain that you control (example doc.qelectrotech.org) so that github can push the files to that adress, but thats for some time in the future.

Le 7 mars 2025 11:05:27 GMT+01:00, Laurent Trinques ***@***.***> a écrit :
>scorpio810 left a comment (qelectrotech/qelectrotech-doc#3)
>
>Hello @Int-Circuit!
>I've compiled the html pages of the manual and made them available here for QET's users to comment on.:
>[Manual_new](https://download.qelectrotech.org/qet/manuals/html/index.html)
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/qelectrotech/qelectrotech-doc/pull/3#issuecomment-2706036680
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

Int-Circuit commented:
Hi, thanks for the feedback, it helps a lot,

The big problem is that the doc dates back to qet 0.7 and that there isn't a lot of explanations about the new  added features and how they work (as someone who is external to the dev of the app)

Could you perhaps give me bulletpoints about 0.8 and 0.9, since 0.10 is still considered beta?

Thanks

Le 7 mars 2025 10:42:55 GMT+01:00, plc-user ***@***.***> a écrit :
>plc-user left a comment (qelectrotech/qelectrotech-doc#3)
>
>Hello @Int-Circuit!
>
>Most of my commits work in the background not to be seen by users, but some are quite visible:
>- implement settable point-size of grid (1 .. 5)
>- element editor: add mirror (shortcut: &lt;M&gt;) and flip (shortcut: &lt;F&gt;) for all primitives
>- element-editor: add rotation with smaller increments (5 degrees) for "line", "polygon", "text" and "dynamic_text" with shortcut &lt;Ctrl&gt;+&lt;Space&gt;
>- element-editor: fix rotation by 90° of primitives (shortcut: &lt;Space&gt;)
>- element-editor: copy-and-paste: set paste-position to meaningful values
>- export: set maximum width / height according limitations in QPainter to 32767 for Pixel-Images and 100000 for svg/dxf (was set to 10000 before globally for all types of export)
>- use QET_ElementScaler from Element-Editor - GUI
>- add "Company-Collection" and "Company-Titleblocks" (can be used like customer-collection and customer-titleblocks)
>
>All these changes are for the version 0.10-dev of QET!
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/qelectrotech/qelectrotech-doc/pull/3#issuecomment-2705985729
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

plc-user commented:
There are some threads in QET-Forum, where differences between versions are noted:
version 0.7 -&gt; 0.8: https://qelectrotech.org/forum/viewtopic.php?id=1953
version 0.8 -&gt; 0.9: https://qelectrotech.org/forum/viewtopic.php?id=2379
My french is not good enough to understand a great part of it.
It seems that these are lists of headlines from github-commits ... what's behind each???

Int-Circuit commented:
Hi @scorpio810, could you please tell me where the source for the Doxygen files are, I am trying to integrate them in the dev docs.

scorpio810 commented:
> Hi @scorpio810, could you please tell me where the source for the Doxygen files are, I am trying to integrate them in the dev docs.

Hi @Int-Circuit source for Doxygen are here:
https://github.com/qelectrotech/qelectrotech-source-mirror/blob/master/Doxyfile

Int-Circuit committed:
fix spelling issues + added page in toc

fixed spelling : final

Int-Circuit committed:
added build steps in doc

Int-Circuit committed:
switched code style setup contrib page

Int-Circuit committed:
added link to doxygen page

removed old code

added branch build

removed need doxygen

Int-Circuit committed:
Removed doxygen.rst since doc being served from differernt page

Int-Circuit commented:
Hello @scorpio810 could you please review the doxygen MR on the QET source repo? 
Thx

Int-Circuit committed:
starting to add ssbewer gist

Int-Circuit committed:
Continuing work on Windows compile instructions

Int-Circuit committed:
update checkout action version (v3 is EoL)

deploy on push

fix for action

switch to ubuntu-latest since 20.04 isn't available anymore

Int-Circuit committed:
added PATH setup

Int-Circuit commented:
@scorpio810
Est ce que tu peux me donner des infos concernant la compil sous windows? Est ce que ce [gist](https://gist.github.com/ssweber/0519e50942926e4a32c22d49b280c0e6#gistcomment-5715738) semble ok?

Int-Circuit committed:
properly fix issue #2

scorpio810 commented:
Hello @Int-Circuit!!

You think this  ok for  merge it this pull request?

Int-Circuit commented:
> Hello @Int-Circuit!!
> 
> You think this ok for merge it this pull request?

Hi @scorpio810, Docs are mostly up to date, only thing I'm not sure is windows manual compile + some env setup but those can be fixed later. Undrafting MR.

Int-Circuit commented:
After that, could u merge the doxygen MR on the source repo? Should be feature complete.

Int-Circuit committed:
added changelog

Int-Circuit committed:
fix version name

remove breathe

added correct codespell skip glob

fix space in spell step

Int-Circuit commented:
> After that, could u merge the doxygen MR on the source repo? Should be feature complete.

wait no I forgot I wanted to change from a push to master to an action that opens a MR so unwanted edits of the QCH file can be droped. Do not merge the doxygen MR, I am going to look at it during the holidays.

Int-Circuit committed:
prepared files for reuse + split actions + implement MR reuse

added testing

fix reuse error

fix commit message

only warn of spell

fail on spell error

fix warn

Int-Circuit commented:
Hi @scorpio810, I've reused the MR logic from doxygen for the licensing step of the github action and consider this MR ready for review. As with the doxygen MR, this will require adding a MR secret to QET-doc so that the action can succeed. 
> WARNING : Since this is a more elaborate setup than a simple push, it requires some setup in the repo settings [^1].
I recommend using a Fine-Grained token so permissions are minimal. This then needs to be added to a secret, in this case MR_TOKEN, but you can change it's name if you will.

Also pls remove any other branch name in the action other than master as this was for testing. If u cannot edit the files pls let me know.

Int-Circuit committed:
- deleting unneeded images
- translated images + translation in future MR

Int-Circuit committed:
ready actions for merge
